CrowdStrike/ember-toucan-core
# #48: Add Checkbox component

## 🚀 Description
This PR introduces a `Checkbox` control component and `CheckboxField` component. 

**NOTE:** These components are not 100% styled yet, as we are still figuring out where to land on a few things. This is about ~75% close though.

Closes #48 

---

## 🔬 How to Test

- These are only presentational components, so test should pass
- View the checkbox link at https://48034b83.ember-toucan-core.pages.dev/docs/components/checkbox and review the documentation
  - **NOTE:** This checkbox does not toggle on click! It is presentational only
- View the link at https://48034b83.ember-toucan-core.pages.dev/docs/components/checkbox-field  and review the documentation
  - **NOTE:** These components are not 100% styled yet, as we are still figuring out where to land on a few things.  This is about ~75% close though.

---

## 📸 Images/Videos of Functionality

**UI States**
<img width="1612" alt="Screenshot 2023-02-24 at 10 12 02 AM" src="https://user-images.githubusercontent.com/8069555/221214784-a11bc413-9851-42dc-8b86-9a59cff4e6d4.png">

**Focus State**

https://user-images.githubusercontent.com/8069555/221663299-f39a228d-c505-4bca-abfb-3f7dddc25ce2.mov

**A11y**

Showing the screenreader picking up:

1. Checkbox state
2. aria-invalid
3. hint text
4. error text
5. hint + error text



https://user-images.githubusercontent.com/8069555/221663069-85e1efe8-6add-4380-8e5f-c169bca8804f.mov
